providers/dns/pdns/pdns.go

```diff
@@ -52,33 +54,39 @@ func NewDNSProviderCredentials(host *url.URL, key string) (*DNSProvider, error)
 		return nil, fmt.Errorf("PDNS API URL missing")
 	}
 
-	provider := &DNSProvider{
+	d := &DNSProvider{
 		host:   host,
 		apiKey: key,
+		client: &http.Client{Timeout: 30 * time.Second},
 	}
-	provider.getAPIVersion()
 
-	return provider, nil
+	var err error
+	d.apiVersion, err = d.getAPIVersion()
```

Comment:
I almost prefer to see something like:

```go
apiVersion, err := d.getAPIVersion()
if err != nil {
  // ...
}

d.apiVersion = apiVersion
```

As it would only "set" the `apiVersion` when there was no error.